lib/pluginmanager/bundler/logstash_injector.rb

```diff
@@ -20,7 +30,12 @@ def self.inject!(new_deps, options = { :gemfile => LogStash::Environment::GEMFIL
       dependencies = new_deps.dependencies.collect(&method(:dependency))
 
       injector = new(bundler_format)
-      injector.inject(gemfile, lockfile, dependencies)
+
+      # Some of the internal classes requires to be inside the LOGSTASH_HOME to find the relative
+      # path of the core gems.
+      Dir.chdir(LogStash::Environment::LOGSTASH_HOME) do
```

Comment:
Yes its block, it will return to the original directory:

```ruby
puts "current dir: #{Dir.pwd}"

Dir.chdir("/Users/ph") do
  puts "current dir: #{Dir.pwd}"
end
puts "current dir: #{Dir.pwd}"

```

```
⚙ph@sashimi/tmp ruby testing.rb
current dir: /private/tmp
current dir: /Users/ph
current dir: /private/tmp
⚙ph@sashimi/tmp
```